### ISSUE TITLE
holland mysql user is granted more privileges than required

### ISSUE THREAD

**Issue by joeashcraft** [rillip3/lamp#11]
In the interest of security, the 'holland' MySQL user should be granted only those permissions which are required for a Holland backup to run.

Privileges are granted on L36 of [mysqldump.yml](https://github.com/rillip3/lamp/blob/master/site-cookbooks/LAMP/files/default/lamp/roles/holland/tasks/mysqldump.yml#L36 "priv=*.*:ALL")

      mysql_user: name=holland host=localhost password="{{mysql_password}}" priv=*.*:ALL

**Comment by joeashcraft** [comment]
tl;dr: holland user should be granted: SELECT, SHOW VIEW, TRIGGER, LOCK TABLES, RELOAD, EVENT.

from [MySQL docs](https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html "mysqldump — A Database Backup Program"):
> mysqldump requires at least the SELECT privilege for dumped tables, SHOW VIEW for dumped views, TRIGGER for dumped triggers, and LOCK TABLES if the --single-transaction option is not used. Certain options might require other privileges as noted in the option descriptions.

## mysqldump option related privs
--routines, -R: SELECT on mysql.proc
 --flush-logs, -F: RELOAD
--master-data: RELOAD; binary logging must be enabled.
--events, -E: EVENT
--triggers: TRIGGER
--single-transaction: negates requirement for LOCK TABLES

## non-option related privs
SELECT: to backup tables
SHOW VIEW: to backup views

## holland monitoring plugin
PROCESS is required for the [Holland plugin](https://github.com/racker/rackspace-monitoring-agent-plugins-contrib/blob/master/holland_mysqldump.py) to allow "mysqladmin status"; though we should use a 'raxmon' user for that instead.

**Comment by joeashcraft** [comment]
I'm working on [modifying](https://github.com/joeashcraft/lamp/pull/1) and testing before submitting a PR here.

**Comment by rillip3** [comment]
Will merge once the PR is tested and submitted
